runtime-infra/fragments/ecs-service.yaml

```diff
@@ -1151,6 +1161,130 @@ Resources:
+              - prefix: "OutOfMemory"
+      State: !Ref MonitorStatus
+      Targets:
+        - Arn: !GetAtt LogGroupOOMEvents.Arn
+          Id: LogTarget
+    
+  # Creating a log group for monitoring the OOM events
+  LogGroupOOMEvents:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Sub "/aws/events/ecs-oom-errors-${MicroServiceUniqueName}"
+      RetentionInDays: 14
+      KmsKeyId:
+        Fn::GetAtt:
+          - LogGroupKmsKey
+          - Arn
+
+  # Resource based policy to allow EventBridge to publish to CloudWatch Logs
+  LogGroupOOMEventsPolicy:
+    Type: AWS::Logs::ResourcePolicy
+    Properties:
+      PolicyName: !Sub "EventBridgeToCWLogsPolicy-${ECSService.Name}"
+      PolicyDocument: !Sub |
+        {
+          "Version": "2012-10-17",
+          "Statement": [
+            {
+              "Sid": "EventBridgetoCWLogsCreateLogStreamPolicy",
+              "Effect": "Allow",
+              "Principal": {
+                "Service": [
+                  "events.amazonaws.com"
+                ]
+              },
+              "Action": [
+                "logs:CreateLogStream"
+              ],
+              "Resource": [
+                "${LogGroupOOMEvents.Arn}"
+              ]
+            },
+            {
+              "Sid": "EventBridgetoCWLogsPutLogEventsPolicy",
+              "Effect": "Allow",
+              "Principal": {
+                "Service": [
+                  "events.amazonaws.com"
+                ]
+              },
+              "Action": [
+                "logs:PutLogEvents"
+              ],
+              "Resource": [
+                "${LogGroupOOMEvents.Arn}"
+              ],
+              "Condition": {
+                "ArnEquals": {"AWS:SourceArn": "${EventRule.Arn}"}
+              }
+            }
+          ]
+        }
+
+  # Create alarm
+  ErrorFatalLogsMetricAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Condition: HasAlarmTopic
+    DependsOn: ErrorFatalLogsMetricFilter
+    Properties:
+      AlarmName: !Sub "${MicroServiceUniqueName}-ErrorFatalLogs-Alarm"
+      AlarmDescription: "CloudWatch alarm for when ECS LogGroup has ERROR or FATAL line."
+      TreatMissingData: notBreaching
+      AlarmActions: 
+        - !Ref AlarmSNSTopicArn
+      #InsufficientDataActions:
+      #  - !Ref AlarmSNSTopicArn
+      OKActions:
+        - !Ref AlarmSNSTopicArn
+      DatapointsToAlarm: 1
+      MetricName: !Sub "${MicroServiceUniqueName}-ErrorFatalMetric"
+      Namespace: "ErrorFatalLogs"
+      ComparisonOperator: GreaterThanOrEqualToThreshold
+      EvaluationPeriods: 60
+      Period: 60
+      Statistic: Sum
+      Threshold: 1
+
+  CloudWatchAlarmOOM:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmName: !Sub "${ECSService.Name}-ECSOutOfMemory-Alarm"
+      AlarmDescription: "CloudWatch alarm for when ECS task stop with OOM errors."
+      TreatMissingData: notBreaching        
+      AlarmActions:
+        - !Ref AlarmSNSTopicArn
```

Comment:
@danilosilvase AlarmSNSTopicArn can be empty, you should add a condition based on `AlarmSNSTopicArn` like we did for `ErrorFatalLogsMetricAlarm`

Reply:
done